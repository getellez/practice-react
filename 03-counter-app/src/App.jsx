import { SayHello } from "./components/SayHello/SayHello";
import { CounterApp } from './components/CounterApp/CounterApp.jsx'

function App() {
  const name = 'Germ√°n'
  return (
    <>
      <CounterApp value={100}/>
      {/* <SayHello name={ name } /> */}
    </>
  )
}

export default App;