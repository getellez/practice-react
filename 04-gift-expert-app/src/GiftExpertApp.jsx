import { useState } from "react"
import { AddCategory } from "./components/AddCategory"
import { GifGrid } from "./components/GifGrid"

export const GiftExpertApp = () => {
  const [categories, setCategories] = useState([])

  const onAddCategory = (inputValue) => {
    if(categories.includes(inputValue)) return;
    setCategories(categories => [inputValue, ...categories])
  }
  
  const onDeleteCategory = (category) => {
    if (categories.includes(category)) {
      const newCategories = categories
      const pos = categories.indexOf(category)
      newCategories.splice(pos,1)
      setCategories([...newCategories])
    }
  }

  return (
    <>
      <h1>GiftExpertApp</h1>

      <AddCategory handleAddCategory={onAddCategory} />

      {
        categories.map((category, index) => (<GifGrid key={category} category={category} handleRemoveCategory={onDeleteCategory}  />))
      }

    </>
  )
}
