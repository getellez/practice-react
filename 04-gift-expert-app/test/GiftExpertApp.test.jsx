import { fireEvent, render, screen } from "@testing-library/react";
import { GiftExpertApp } from "../src/GiftExpertApp";

jest.mock('../src/GiftExpertApp')

describe('<GiftExpertApp> Component', () => {
  test('should match thâˆ«e snapshot', () => {
    const { container } = render(<GiftExpertApp />)
    expect(container).toMatchSnapshot()
  });
  test('should have the correct title', () => {
    const title = 'GiftExpertApp'
    render(<GiftExpertApp />)

    expect(screen.getByText(title)).toBeTruthy()
  });
  test('should do something', () => {
    const category = 'One Punch Man'
    render(<GiftExpertApp />)
    const form = screen.getByRole('form')
    fireEvent.submit(form)
    const spy = jest.spyOn(GiftExpertApp, 'onAddCategory')
    expect(spy).toHaveBeenCalled()
  });
})