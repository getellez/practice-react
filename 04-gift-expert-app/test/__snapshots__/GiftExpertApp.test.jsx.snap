// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiftExpertApp> Component should match the snapshot 1`] = `
<div>
  <h1>
    GiftExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      autocomplete="off"
      name="categoryInput"
      placeholder="Buscar GIFs"
      type="text"
      value=""
    />
  </form>
</div>
`;

exports[`<GiftExpertApp> Component should match thâˆ«e snapshot 1`] = `
<div>
  <h1>
    GiftExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      autocomplete="off"
      name="categoryInput"
      placeholder="Buscar GIFs"
      type="text"
      value=""
    />
  </form>
</div>
`;
